src/HDF5.jl

```diff
@@ -1952,6 +1820,15 @@ function hdf5_to_julia_eltype(objtype)
     return T
 end
 
+function get_jl_type(objtype)
```

Comment:
Not sure I understand the point of this method.

The `hdf5_to_julia_eltype` method 
already has defined `class_id == H5T_OPAQUE` 
https://github.com/JuliaIO/HDF5.jl/blob/ece0dcd35e8578231a80e8315d76754fce206945/src/HDF5.jl#L1808

So it seems that without this method things should just work already.

Reply:
Yeah this mostly duplicates `hdf5_to_julia_eltype` but changes the string handling part so there are no special cases for the length=1 case. I wrote a new function instead of changing the original because I wanted to only touch the read part of the library. There is probably some consolidation/cleanup possible.